examples/statistics/confidence_ellipse.py

```diff
@@ -0,0 +1,220 @@
+"""
+======================================================
+Plot a confidence ellipse of a two-dimensional dataset
+======================================================
+
+This example shows how to plot a confidence ellipse of a
+two-dimensional dataset, using its pearson correlation coefficient.
+
+The approach that is used to obtain the correct geometry is
+explained and proved here:
+
+https://carstenschelp.github.io/2018/09/14/Plot_Confidence_Ellipse_001.html
+
+The method avoids the use of an iterative eigen decomposition algorithm
+and makes use of the fact that a normalized covariance matrix (composed of
+pearson correlation coefficients and ones) is particularly easy to handle.
+"""
+
+#############################################################################
+#
+# The plotting function itself
+# """"""""""""""""""""""""""""
+#
+# This function plots the confidence ellipse of the covariance of the given
+# array-like variables x and y. The ellipse is plotted into the given
+# axes-object ax.
+#
+# The radiuses of the ellipse can be controlled by n_std which is the number
+# of standard deviations. The default value is 3 which makes the ellipse
+# enclose 99.7% of the points (given the data is normally distributed
+# like in these examples).
+
+
+import numpy as np
+import matplotlib.pyplot as plt
+from matplotlib.patches import Ellipse
+import matplotlib.transforms as transforms
+
+
+def confidence_ellipse(x, y, ax, n_std=3.0, **kwargs):
```

Comment:
```suggestion
def confidence_ellipse(x, y, ax, n_std=3.0, facecolor='none', **kwargs):
```
Confidence ellipses will mostly be drawn without facecolor, and even if they get a facecolor it should most likely not be the bold default colors. So it makes sense to remove the facecolor by default. Will also save you from writing `facecolor='none'` in almost every use of `confidence_ellipse`.

Don't forget to pass `facecolor` on to `Ellipse()` if you change this :smiley:.

Reply:
@timhoffm, @QuLogic : Thanks guys, nice improvements - I will process them shortly. Very busy, these days.